contracts/ERC1155M.sol

```diff
@@ -130,6 +143,21 @@ contract ERC1155M is IERC1155M, ERC1155Supply, ERC2981, Ownable2Step, Reentrancy
         _authorizedMinters[minter] = false;
     }
 
+    /**
+     * @dev Returns cosign nonce.
+     */
+    function getCosignNonce(address minter, uint256 tokenId) public view returns (uint256) {
+        return totalMintedByAddress(minter)[tokenId];
```

Comment:
The implementation for `totalMintedByAddress` relies on a loop. Could this value be tracked as a uint that gets incremented on mint instead? It looks like the loop is capped by a constant value, is that to prevent running out of gas? Then does that mean an address can only mint so many token ids and the user must change to a new account?

Reply:
Yeah good call. I think we can use single mapping `mapping(address => uint32)` (total tokens minted per wallet). This may save some gas (pending experiment). I can test it in following up PR.

The cap is `NUM_TOKENS`. It's the number of "editions" of a ERC1155 collection. It's fixed / determined in constructor. Each edition can has unlimited copies.